src/users/users.service.ts

```
@@ -202,4 +209,55 @@ export class UsersService {
       id: user.id,
     };
   }
+
+  async forgotPassword(email: string): Promise<Record<string, unknown>> {
+    try {
+      const user = await this.userRepository.findOne({
+        where: { email: email, status: UserStatus.ACTIVE },
+      });
+      const status = await this.emailVerificationService.getStatus(email);
+
+      if (status !== EmailVerificationStatus.VERIFIED) {
+        throw new Error();
+      }
+
+      const payload = { userEmail: user.email };
+      const token = this.jwtService.sign(payload, { secret: user.password });
+
+      await this.emailService.emailUser(
+        user.email,
+        passwordResetSubject,
+        resetPasswordBody(token, user.id),
+      );
+
+      return {};
+    } catch (error) {
+      throw new NotFoundException(MSG.EMAIL_NOT_FOUND);
```

You are assuming here that all the possible errors are caused of wrong email. That's a bit of  a stretch. I can think different reasons for this to fail, yet here we are swallowing the original error and just returning 404

---

I did that intentionally for security reasons. I think that here we should return only the generic message, without any explicit explanation what happened.?